onmt/data/SampledDataset.lua

```diff
@@ -0,0 +1,224 @@
+--[[ Data management and batch creation. Handles data created by `preprocess.lua`. ]]
+require 'torchx'
+
+local SampledDataset = torch.class("SampledDataset")
+
+--[[ Initialize a data object given aligned tables of IntTensors `srcData`
+  and `tgtData`.
+--]]
+function SampledDataset:__init(srcData, tgtData, samplingSize, sample_w_ppl, sample_w_ppl_init, sample_w_ppl_max)
+
+  self.src = srcData.words
```

Comment:
@Jungi - in #142, I introduce the ability to replace word input by direct feature input (srcData.vectors) - here your code fully assume that it is words, can you anticipate this change and make your code more generic?